app/MindWork AI Studio/Chat/ChatThread.cs

```diff
@@ -121,6 +121,11 @@ public string PrepareSystemPrompt(SettingsManager settingsManager, ChatThread ch
             }
         }
         
+        // We need a way to save the changed system prompt in our chat thread.
+        // Otherwise, the chat thread will always tell us that it is using the
+        // default system prompt:
+        chatThread = chatThread with { SystemPrompt = systemPromptTextWithChatTemplate };
+        
```

Comment:
Let's think this through again and discuss if we actually need the system prompt in the chat thread, or if we should just delete it since it's generated dynamically every time anyway...